ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Connection.scala

```diff
@@ -3,16 +3,79 @@
 
 package com.daml.http.dbbackend
 
-import cats.effect.{ContextShift, IO}
+import cats.effect.{Blocker, ContextShift, IO}
+import com.daml.http.JdbcConfig
+import com.zaxxer.hikari.{HikariConfig, HikariDataSource}
 import doobie._
 
+import java.io.Closeable
+import java.util.concurrent.Executors.newWorkStealingPool
+import javax.sql.DataSource
+import scala.concurrent.ExecutionContext
+
 object Connection {
 
   type T = Transactor.Aux[IO, Unit]
 
-  def connect(jdbcDriver: String, jdbcUrl: String, username: String, password: String)(implicit
+  def connect(cfg: JdbcConfig)(implicit
       cs: ContextShift[IO]
   ): T =
     Transactor
-      .fromDriverManager[IO](jdbcDriver, jdbcUrl, username, password)(IO.ioConcurrentEffect(cs), cs)
+      .fromDriverManager[IO](cfg.driver, cfg.url, cfg.user, cfg.password)(
+        IO.ioConcurrentEffect(cs),
+        cs,
+      )
+}
+
+object ConnectionPool {
+
+  /*
+  TODO:
```

Comment:
Added TODO's to refactor duplicate code and add configurable props, which I'll get to in a follow-up PR